config/webpack/webpack.config.js

```diff
@@ -170,6 +172,14 @@ const webpackConfig = (options, env, argv) => {
   });
 };
 
+const basepathConfig = {
+  plugins: [
+    new webpack.DefinePlugin({
+      BASEPATH: JSON.stringify(base),
+    }),
+  ],
+};
+
 const defaultWebpackConfig = (env = {}, argv = {}) => {
   const { disableAggregateTranslations, disableHotReloading } = env;
```

Comment:
I'm thinking we should make this change by adding an additional webpack env to follow syntax we've done with other webpack envs to toggle configuration. Something like:

```
const { basepath, disableAggregateTranslations, disableHotReloading } = env;

...

 return webpackConfig(options, env, argv);
```

Then we can compile webpack like:

```bash
webpack --config webpack.config.js -p --env.basepath='mybaseroot/'
```

We should then have access to add this directly to the plugins listed without merging the default configuration